src/start_service.ts

```diff
@@ -17,13 +18,15 @@ export async function startService(opts: {
     ? { encryptionPublicKey: web3Interface.encryptionWallet.publicKey }
     : validateEnvVars();
 
-  const { nodeName, bootstrapList, listenAddresses, peerId } = opts;
+  let { nodeName, bootstrapList, listenAddresses, peerId } = opts;
+
+  peerId = peerId ?? await loadPeerIdFromFile();
 
   const arch = new Archaeologist({
     name: nodeName,
     bootstrapList: bootstrapList ?? process.env.BOOTSTRAP_LIST?.split(",").map(s => s.trim()),
```

Comment:
you haven't touched this but where is bootstrap list coming from?

Reply:
This is somewhat stale code -- the bootstrap list has for the most part been removed from the code, and this can probably get removed as well (will make a task for this).